File: core/src/main/java/ir/logicfan/core/util/Connectivity.java
package ir.logicfan.core.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import javax.inject.Singleton;

/**
 * Check device's network connectivity
 */
@Singleton
public class Connectivity {

    public static boolean isUserOnline(Context context) {
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo netInfo = connectivityManager != null ? connectivityManager.getActiveNetworkInfo() : null;
        return (netInfo != null && netInfo.isConnected());
    }
}

File: core/src/main/java/ir/logicfan/core/util/Connectivity.kt
package ir.logicfan.core.util

import android.content.Context
import android.net.ConnectivityManager

/**
 * Check device's network connectivity
 */
object Connectivity {

    fun isUserOnline(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return connectivityManager.activeNetworkInfo.isConnected
    }
}